docker/resource_docker_plugin_funcs.go

```diff
@@ -0,0 +1,165 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func getDockerPluginArgs(src interface{}) []string {
+	if src == nil {
+		return nil
+	}
+	b := src.(*schema.Set)
+	args := make([]string, b.Len())
+	for i, a := range b.List() {
+		args[i] = a.(string)
+	}
+	return args
+}
+
+func resourceDockerPluginCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).DockerClient
+	ctx := context.Background()
+	pluginRef := d.Get("plugin_reference").(string)
+	alias := d.Get("alias").(string)
+	log.Printf("[DEBUG] Install a Docker plugin " + pluginRef)
+	body, err := client.PluginInstall(ctx, alias, types.PluginInstallOptions{
+		RemoteRef:            pluginRef,
+		AcceptAllPermissions: d.Get("grant_all_permissions").(bool),
+		Disabled:             d.Get("disabled").(bool),
+		Args:                 getDockerPluginArgs(d.Get("args")),
+	})
+	if err != nil {
+		return fmt.Errorf("install a Docker plugin "+pluginRef+": %w", err)
+	}
+	_, _ = ioutil.ReadAll(body)
+	key := pluginRef
+	if alias != "" {
+		key = alias
+	}
+	plugin, _, err := client.PluginInspectWithRaw(ctx, key)
+	if err != nil {
+		return fmt.Errorf("inspect a Docker plugin "+key+": %w", err)
+	}
+	setDockerPlugin(d, plugin)
+	return nil
+}
+
+func setDockerPlugin(d *schema.ResourceData, plugin *types.Plugin) {
+	d.SetId(plugin.ID)
+	d.Set("plugin_reference", plugin.PluginReference)
+	d.Set("alias", plugin.Name)
+	d.Set("disabled", !plugin.Enabled)
+	// TODO support other settings
+	// https://docs.docker.com/engine/reference/commandline/plugin_set/#extended-description
+	// source of mounts .Settings.Mounts
+	// path of devices .Settings.Devices
+	// args .Settings.Args
+	d.Set("args", plugin.Settings.Env)
```

Comment:
Env is not the correct value for Args.
You need to implement all of https://pkg.go.dev/github.com/docker/docker/api/types#PluginSettings

Reply:
I misunderstood the purpose of this value. You might want to change the attribute to `env` to make clear what it is doing.